### DISCUSSION TITLE
Add Advanced Filter to drawer select

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#6892] [Feature Requests]
Described [here](https://github.com/directus/directus/discussions/6097#discussioncomment-835038). The referenced discussion also describes a typical use case, but there are more, especially with Directus files.

I took a look at the code, but lack knowledge about Directus & Vue to lift that on my own without guidance or best practices of the code base. Also @benhaynes asked for a bigger discussion on that idea.

**Comment by benhaynes** [comment]
I think this is an option to consider, we basically need to see what a design would look like and compare that to what we have now (in the sidebar). Either option could be added into the drawer, but adding the sidebar to drawers seems messier.

**Reply by nickluger** [reply]
Just a quick draft ✍️:

### Option 1: Filter in sidebar on top of (right of) drawer

![Sidebar Filter in Drawer](https://user-images.githubusercontent.com/32647249/126449087-a3b33888-0491-4274-9d78-4f2f9760ee2f.jpeg)

Mobile or on narrow screens this might become a bit messier with two modals/overlays above each other. But i don't know to what extent Directus supports mobile access. Everything just feels pretty responsive!

![Sidebar Filter in Drawer Mobile](https://user-images.githubusercontent.com/32647249/126449243-ff270354-38d1-476b-bec9-ff490f37a809.jpeg)

### Option 2: Filter in dropdown menu next to sort-buttons:

![Filter in Dropdown menu](https://user-images.githubusercontent.com/32647249/126449459-7d48db17-a98e-4098-80e7-c51b669b5d03.jpeg)

The dropdown menu is already used with the sort button:

![Bildschirmfoto 2021-07-21 um 09 40 05](https://user-images.githubusercontent.com/32647249/126450538-7ddee65b-b5f7-4ead-a9d8-584000277f50.png)

This filter button does not have to be on the right of the other buttons (sort, cardsize etc.) but could also be left of them.

On mobile the space is a bit constrained for an additional button in that button bar, but an icon-only button would fit in.

![Filter in Dropdown menu  mobile](https://user-images.githubusercontent.com/32647249/126451883-dc8acf2a-db9e-4760-a996-144d02f3ab89.jpeg)

### Option 3: Somewhere next to the search button/bar

![Filter next to search](https://user-images.githubusercontent.com/32647249/126453301-26b87b0d-83ff-4972-85b3-b308d92843dd.jpeg)

As the search bar extends to the left, the button could be right of the search bar.


Please let me know what you think! ☺️

**Reply by rijkvanzanten** [reply]
I think adding the full sidebar might be a little overkill for just the filter. I don't wanna put it in the layout itself (the gray "bar" underneath), as that is different from layout to layout and might not be there. I think option 3 is the best 🙂

**Reply by benhaynes** [reply]
I agree. Most of these sidebar features don't need to crowd the drawer. I think the only key one is Advanced Filters... which can go in or beside the search bar (like in v8) as seen in your Option 3.

EDIT: This will also give us more horizontal room for the filters... which would be nice.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
